lsp-ui-sideline.el

```diff
@@ -563,8 +565,7 @@ from the language server."
            symbols)
       (setq lsp-ui-sideline--tag tag
             lsp-ui-sideline--last-width (window-text-width))
-      (when (and line-changed lsp-ui-sideline-show-diagnostics)
-        (lsp-ui-sideline--diagnostics buffer bol eol))
+      (lsp-ui-sideline--diagnostics buffer bol eol)
```

Comment:
Remove the check `line-changed` or else if the user scroll horizontally this doesn't take effect.

Function `lsp-ui-sideline--diagnostics` already check for the flag `lsp-ui-sideline-show-diagnostics` hence we simply remove the entire check here.